### PR TITLE
Test proxy

### PR DESCRIPTION
**Pull request by gearama**
Replaces the homegrown test-proxy with the commn one used by other languages, created byt the ENG team . (https://github.com/Azure/azure-sdk-tools/tree/main/tools/test-proxy )
One bebefit is that the test recordings are kept in a separate repo ( https://github.com/Azure/azure-sdk-assets) from where the tests are restored at playback time. 
All recordinggs have been moved to the new location. A few misbehaving tests(<10) have been switched to LIVEONLY , a bug has been filed for those to keep track
To install the proxy locally and run it read : https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md#installation 

or execute "\eng\scripts\Start-TestProxy.ps1 -AssetsPath <sdk/service> "

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: cmake-modules/TestProxyPrep.cmake [hunk 1]
```diff
@@ -0,0 +1,6 @@
+macro(CopyTestProxyScripts)
+    file(COPY ${CMAKE_SOURCE_DIR}/eng/Scripts/Start-TestProxy.ps1
+        DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+    file(COPY ${CMAKE_SOURCE_DIR}/eng/Scripts/Stop-TestProxy.ps1
+        DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+endmacro()
```
Comment (RickWinter) [inline cmake-modules/TestProxyPrep.cmake:6]:
```suggestion
endmacro()

```

File: eng/scripts/Start-TestProxy.ps1 [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    # if we have errors this means we had issues installing it , needs to be done by hand
+    if($error){
+        echo "unable to install testproxy. try manual install"
+        exit 1
+    }
+}
+
+echo "start test proxy with argument list --storage-location $AssetsPath"
+#starts it in a separate process that will outlive pwsh in order to serve requests.
+Start-Process 'test-proxy' -ArgumentList "--storage-location $AssetsPath"
```
Comment (RickWinter) [inline eng/scripts/Start-TestProxy.ps1:39]:
```suggestion
Start-Process 'test-proxy' -ArgumentList "--storage-location $AssetsPath" 

```

File: eng/scripts/Stop-TestProxy.ps1 [hunk 3]
```diff
@@ -0,0 +1 @@
+﻿Stop-Process -Name "test-proxy"
```
Comment (RickWinter) [inline eng/scripts/Stop-TestProxy.ps1:1]:
```suggestion
﻿Stop-Process -Name "test-proxy"
﻿
```

Comment (RickWinter) [inline eng/scripts/Stop-TestProxy.ps1:1]:
```suggestion
# Copyright (c) Microsoft Corporation. All rights reserved.
# SPDX-License-Identifier: MIT

﻿Stop-Process -Name "test-proxy"
```

File: eng/scripts/Start-TestProxy.ps1 [hunk 4]
```diff
@@ -0,0 +1,39 @@
+param(
```
Comment (RickWinter) [inline eng/scripts/Start-TestProxy.ps1:1]:
```suggestion
# Copyright (c) Microsoft Corporation. All rights reserved.
# SPDX-License-Identifier: MIT

param(
```

File: sdk/core/azure-core-test/CMakeLists.txt [hunk 5]
```diff
@@ -10,28 +10,25 @@ set(CMAKE_CXX_STANDARD_REQUIRED True)
...
   src/test_base.cpp
+  src/test_proxy_manager.cpp
 )
 
 add_library (
   azure-core-test-fw
   ${AZURE_CORE_TEST_HEADER}
   ${AZURE_CORE_TEST_SOURCE}
-  )
+   "src/test_proxy_manager.cpp" "src/test_proxy_policy.cpp")
```
Comment (RickWinter) [inline sdk/core/azure-core-test/CMakeLists.txt:31]: This is unnessecary.  These two files are part of AZURE_CORE_TEST_SOURCE

File: sdk/core/azure-core-test/inc/azure/core/test/test_proxy_manager.hpp [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+
+  private:
+    std::string PrepareRequestBody();
+    void StartPlaybackRecord(TestMode testMode);
+    void StopPlaybackRecord();
+    void SetProxySanitizer();
+    bool CheckSanitizers();
+  };
+
+}}} // namespace Azure::Core::Test
```
Comment (RickWinter) [inline sdk/core/azure-core-test/inc/azure/core/test/test_proxy_manager.hpp:179]:
```suggestion
}}} // namespace Azure::Core::Test

```

File: sdk/core/azure-core-test/inc/azure/core/test/test_proxy_policy.hpp [hunk 7]
```diff
@@ -11,52 +11,40 @@
...
   public:
     /**
      * @brief Disable default constructor.
      *
      */
-    RecordNetworkCallPolicy() = delete;
+    TestProxyPolicy() = delete;
 
     /**
      * @brief Construct the record network policy which will save the HTTP request and response to
```
Comment (RickWinter) [inline sdk/core/azure-core-test/inc/azure/core/test/test_proxy_policy.hpp:42]:
```suggestion
     * @brief Construct the TestProxyPolicy which will save the HTTP request and response to
```

File: sdk/core/azure-core-test/inc/azure/core/test/test_proxy_policy.hpp [hunk 8]
```diff
@@ -11,52 +11,40 @@
...
+     * @param testProxy A reference to the test proxy manager
      */
-    RecordNetworkCallPolicy(Azure::Core::Test::InterceptorManager* interceptorManager)
-        : m_interceptorManager(interceptorManager), m_symbol(std::make_unique<uint8_t>('x'))
-    {
-    }
+    TestProxyPolicy(Azure::Core::Test::TestProxyManager* testProxy) : m_testProxy(testProxy) {}
 
     /**
      * @brief Cronstructs a new record network policy with the same recorded data.
```
Comment (RickWinter) [inline sdk/core/azure-core-test/inc/azure/core/test/test_proxy_policy.hpp:50]:
```suggestion
     * @brief Cronstructs a new TestProxyPolicy with the same recorded data.
```

File: sdk/core/azure-core-test/inc/azure/core/test/test_proxy_policy.hpp [hunk 9]
```diff
@@ -83,4 +71,4 @@ namespace Azure { namespace Core { namespace Test {
         const Azure::Core::Context& ctx) const override;
   };
 
-}}} // namespace Azure::Core::Test
+}}} // namespace Azure::Core::Test
```
Comment (RickWinter) [inline sdk/core/azure-core-test/inc/azure/core/test/test_proxy_policy.hpp:74]:
```suggestion
}}} // namespace Azure::Core::Test

```

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_delete_test_live.cpp [hunk 10]
```diff
@@ -110,7 +102,6 @@ TEST_F(KeyVaultKeyClient, DeleteInvalidKey)
         static_cast<typename std::underlying_type<Azure::Core::Http::HttpStatusCode>::type>(
             error.StatusCode),
         404);
-    EXPECT_EQ(error.Message, GetNotFoundErrorMsg(keyName));
```
Comment (RickWinter) [inline sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_delete_test_live.cpp:113]: Why did you remove this check?

File: sdk/storage/ci.yml [hunk 11]
```diff
@@ -27,8 +27,8 @@ stages:
       ServiceDirectory: storage
       # Not running tests on CI but only on LiveTest mode
       CtestRegex: azure-storage
-      LineCoverageTarget: 73.5305
-      BranchCoverageTarget: 40.7738
+      LineCoverageTarget: 71.0434
+      BranchCoverageTarget: 40.3229
```
Comment (ahsonkhan) [inline sdk/storage/ci.yml:31]: Ideally, we shouldn't be losing coverage here.

**Review by ahsonkhan — COMMENTED** [review]
Some comments and questions to try and understand some of the rationale for changes made here.

File: cmake-modules/TestProxyPrep.cmake [hunk 12]
```diff
@@ -0,0 +1,6 @@
+macro(CopyTestProxyScripts)
```
Comment (RickWinter) [inline cmake-modules/TestProxyPrep.cmake:1]:
```suggestion
# Copyright (c) Microsoft Corporation. All rights reserved.
# SPDX-License-Identifier: MIT

macro(CopyTestProxyScripts)
```

File: .vscode/cspell.json [hunk 13]
```diff
@@ -25,7 +25,8 @@
     "*nlohmann-json*",
     "eng/docs/api/assets/**/*",
     "eng/CredScanSuppression.json",
-    "**/test/ut/recordings/*.json"
+    "**/test/ut/recordings/*.json",
+    "**/test/ut/test_base.hpp"
```
Comment (RickWinter) [inline .vscode/cspell.json:29]: This file is part of the source.  It needs to be scanned as well.
```suggestion
```

File: eng/scripts/Stop-TestProxy.ps1 (hunk 3 above)
Comment (ahsonkhan) [reply eng/scripts/Stop-TestProxy.ps1:1]: I was going to give the same feedback, but turns out majority (if not all) of the scripts in eng/scripts lack the license header.

I was told it wasn't necessary to add them there.

cc @danieljurek, @weshaggard 

Same in other repos like Embedded C:
https://github.com/Azure/azure-sdk-for-c/pull/2304#discussion_r943813969

File: .vscode/cspell.json (hunk 13 above)
Comment (ahsonkhan) [reply .vscode/cspell.json:29]: Jinx (same comment as the following):
https://github.com/Azure/azure-sdk-for-cpp/pull/4118#discussion_r1044912974

File: sdk/attestation/azure-security-attestation/test/ut/policygetset_test.cpp [hunk 14]
```diff
@@ -405,6 +405,7 @@ namespace Azure { namespace Security { namespace Attestation { namespace Test {
     std::string GetTestName(testing::TestParamInfo<PolicyTests::ParamType> const& testInfo)
     {
       std::string testName;
+      int suffixVotes = 0;
```
Comment (RickWinter) [inline sdk/attestation/azure-security-attestation/test/ut/policygetset_test.cpp:408]: If I understand the logic correctly, We only run ModifyPolicySecured on AAD in live mode.

Thus you could instead write the following code which is equivalent and easier to understand:
```
if(testInfo.param.InstanceType == ServiceInstanceType::AAD 
&& testInfo.param.TestType == TestCaseType::ModifyPolicySecured) {
    testName += "_LIVEONLY_";
}
```

File: sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp [hunk 15]
```diff
@@ -388,5 +404,6 @@ namespace Azure { namespace Core { namespace Test {
      *
      */
     void TearDown() override;
+    virtual std::string GetAssetsPath() { return "assets.json"; }
```
Comment (RickWinter) [inline sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp:407]: Add a brief for this method

File: sdk/core/azure-core-test/inc/azure/core/test/test_context_manager.hpp [hunk 16]
```diff
@@ -20,6 +20,11 @@ namespace Azure { namespace Core { namespace Test {
    */
   class TestContextManager {
   public:
+    /**
+     * @brief The path where the asset.json exists.
+     *
+     */
+    std::string AssetsPath;
```
Comment (RickWinter) [inline sdk/core/azure-core-test/inc/azure/core/test/test_context_manager.hpp:27]: Why is the AssetsPath on both the TestBase and the TestContextManager?  Can you add more info to the brief explaning how this value flows between the two.

**Review by RickWinter — COMMENTED** [review]
Added some feedback.  I'll review again once addressed.

File: sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp [hunk 17]
```diff
@@ -7,21 +7,19 @@
...
 
-#include "azure/core/test/interceptor_manager.hpp"
 #include "azure/core/test/network_models.hpp"
 #include "azure/core/test/test_context_manager.hpp"
 #include <azure/core/credentials/credentials.hpp>
 #include <azure/core/credentials/token_credential_options.hpp>
 #include <azure/core/internal/client_options.hpp>
 #include <azure/core/internal/diagnostics/log.hpp>
 #include <azure/core/internal/environment.hpp>
+#include <azure/core/test/test_proxy_manager.hpp>
```
[the remark this answers is not in the document]
Comment (antkmsft) [reply sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp:17]: I'm surprised if Clang-format does not error here in CI. It should.

**Review by antkmsft — APPROVED** [review]
Thank you for the ton of work you have done here. I am glad to see these recording files gone from the source tree. This is a great change.
Other guys have left pretty good feedback - suggestions and questions. I agree with them.
Once they are resolved/applied/answered, I see no blockers to merge this PR.

**Review by Jinming-Hu — APPROVED** [review]
The changes to storage looks good to me.

We can look into the disabled test cases later. I don't want to block this PR.

File: sdk/core/azure-core-test/CMakeLists.txt (hunk 5 above)
Comment (gearama) [reply sdk/core/azure-core-test/CMakeLists.txt:31]: auto added by VS

File: sdk/core/azure-core-test/inc/azure/core/test/test_proxy_policy.hpp (hunk 7 above)
Comment (gearama) [reply sdk/core/azure-core-test/inc/azure/core/test/test_proxy_policy.hpp:42]: done

File: sdk/core/azure-core-test/inc/azure/core/test/test_proxy_policy.hpp (hunk 8 above)
Comment (gearama) [reply sdk/core/azure-core-test/inc/azure/core/test/test_proxy_policy.hpp:50]: done

File: sdk/core/azure-core-test/inc/azure/core/test/test_proxy_policy.hpp (hunk 9 above)
Comment (gearama) [reply sdk/core/azure-core-test/inc/azure/core/test/test_proxy_policy.hpp:74]: done

File: .gitignore [hunk 18]
```diff
@@ -346,3 +346,4 @@ build/
 
 # Default Assets restore directory
 .assets
+/.cache/clangd/index
```
[the remark this answers is not in the document]
Comment (gearama) [reply .gitignore:349]: this was added by the test proxy some temp folder used when uploading, removed, it, it will be reinserted when needed

File: cmake-modules/TestProxyPrep.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (gearama) [reply cmake-modules/TestProxyPrep.cmake]: done

File: eng/scripts/Start-TestProxy.ps1 [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+$running = Get-Process -Name test-proxy
+echo $AssetsPath
+if($running)
+{
+    echo "test-proxy running, no need for new instance"
+    exit 0
+}
+
+# make sure errors collection is empty
+$error.clear()
```
[the remark this answers is not in the document]
Comment (gearama) [reply eng/scripts/Start-TestProxy.ps1:15]: they are displayed by the Get-Process thingy , but they are left in the error collection nonetheless ,

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_delete_test_live.cpp (hunk 10 above)
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-keys/test/ut/key_client_delete_test_live.cpp:113]: remeber we had this discussion regarding matching strings, the KV decided to update the message which would now cause the test to fail ( e.g. add a comma to the string )

File: sdk/core/azure-core-test/src/test_proxy_manager.cpp [hunk 20]
```diff
@@ -0,0 +1,309 @@
...
+
+  return body;
+}
+
+void TestProxyManager::SetStartRecordMode()
+{
+  if (IsPlaybackMode() || IsRecordMode())
+  {
+    std::string mode = (IsPlaybackMode() ? "playback" : "record");
+    // throw std::runtime_error();
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/core/azure-core-test/src/test_proxy_manager.cpp:98]: done

File: sdk/keyvault/azure-security-keyvault-keys/test/ut/key_rotation_policy_test_live.cpp [hunk 21]
```diff
@@ -81,7 +81,7 @@ TEST_F(KeyVaultKeyClient, GetKeyRotationPolicy)
   }
 }
 
-TEST_F(KeyVaultKeyClient, GetRandomBytes)
+TEST_F(KeyVaultKeyClient, GetRandomBytes_LIVEONLY_)
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/keyvault/azure-security-keyvault-keys/test/ut/key_rotation_policy_test_live.cpp:84]: when i don't want it to run from a recording , only when in LIVE mode, this test i marked it as such due to issues in getting a reliable recording using the test proxy, there are ~8 in storage e and 2-3 in the other SDKs hat require further investigation to get working properly with e new infrastructure.

File: sdk/storage/ci.yml (hunk 11 above)
Comment (gearama) [reply sdk/storage/ci.yml:31]: true, but because i had to remove a few tests from the UT suite it did drop a bit. will be back up once we fix the failing tests

File: cmake-modules/TestProxyPrep.cmake (hunk 12 above)
Comment (gearama) [reply cmake-modules/TestProxyPrep.cmake:1]: done

File: vcpkg.json [hunk 22]
```diff
@@ -21,6 +21,9 @@
     {
       "name": "wil",
       "platform": "windows"
+    },
+    {
+      "name": "gtest"
```
[the remark this answers is not in the document]
Comment (gearama) [reply vcpkg.json:26]: oops, good catch , the diff won't work for me for all the files , in any case, leftover from whe i was trying to get gtest working

File: .vscode/cspell.json (hunk 13 above)
Comment (gearama) [reply .vscode/cspell.json:29]: done

File: eng/scripts/Start-TestProxy.ps1 [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+
+#check if we have a test-proxy available
+$CurrentVersion = (Get-Command -Name "test-proxy" -ErrorAction SilentlyContinue).Version
+
+if($error){
```
[the remark this answers is not in the document]
Comment (gearama) [reply eng/scripts/Start-TestProxy.ps1:20]: we cleared $ error , if getcommand fails a new error would be added in to $error , thus we know an error hapepned during tthe previous cmdlet execution , which means you don't have test proxy installed , thus we will attempt to install it for you

File: eng/scripts/Start-TestProxy.ps1 [hunk 24]
```diff
@@ -0,0 +1,39 @@
...
+    echo "Will install testproxy"
+
+    dotnet tool update azure.sdk.tools.testproxy --global --add-source https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json --version "1.0.0-dev*"
```
[the remark this answers is not in the document]
Comment (gearama) [reply eng/scripts/Start-TestProxy.ps1:23]: yes, that is the command from the test-proxy documentation,

File: eng/scripts/Start-TestProxy.ps1 [hunk 25]
```diff
@@ -0,0 +1,39 @@
...
+    # clear the errors again
+    $error.clear()
+
+    #check again for test proxy presence
+    $CurrentVersion = (Get-Command -Name "test-proxy" -ErrorAction SilentlyContinue).Version
+
+    # if we have errors this means we had issues installing it , needs to be done by hand
+    if($error){
+        echo "unable to install testproxy. try manual install"
```
[the remark this answers is not in the document]
Comment (gearama) [reply eng/scripts/Start-TestProxy.ps1:32]: go through the test proxy docs, figure out if you're in a magic scenario, maybe you need some other flags passed in , maybe dotnet needs some missing component, or needs to be reinstalled, or a newwer version, no idea what you might need to do . the output from executing those commands you help with diagnosing.

[the remark this answers is not in the document]
Comment (gearama) [reply eng/scripts/Start-TestProxy.ps1:32]: done

File: eng/scripts/Stop-TestProxy.ps1 (hunk 3 above)
[the remark this answers is not in the document]
Comment (gearama) [reply eng/scripts/Stop-TestProxy.ps1]: done

File: sdk/attestation/assets.json [hunk 26]
```diff
@@ -0,0 +1,6 @@
+{
+  "AssetsRepo": "Azure/azure-sdk-assets",
+  "AssetsRepoPrefixPath": "cpp",
+  "TagPrefix": "cpp/attestation",
+  "Tag": "cpp/attestation_b384d96f95"
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/attestation/assets.json:5]: that is autogenerated by the test proxy on upload , not to be manually edited, will be updated on next upload. 
The test proxy uses tags/releases to keep track of files and versions

File: eng/scripts/Start-TestProxy.ps1 (hunk 2 above)
[the remark this answers is not in the document]
Comment (gearama) [reply eng/scripts/Start-TestProxy.ps1]: nope, that is for the developer's convenience, the proxy is started stopped in the piplines by some magic set of jobs

[the remark this answers is not in the document]
Comment (gearama) [reply eng/scripts/Start-TestProxy.ps1]: we can remove them if we don't like it

File: sdk/attestation/azure-security-attestation/test/ut/policygetset_test.cpp [hunk 27]
```diff
@@ -440,6 +443,11 @@ namespace Azure { namespace Security { namespace Attestation { namespace Test {
 
       testName += "_";
       testName += testInfo.param.TeeType.ToString();
+      if (suffixVotes == 2)
+      {
+        testName += "_LIVEONLY_";
+      };
+      //+"_LIVEONLY_";
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/attestation/azure-security-attestation/test/ut/policygetset_test.cpp:450]: because in attestetion the UTs are overengineered , instead of having like everywhere else TEST(vv,ddd) someone decided that they can have one generic method and they can dinamically update the name of the test at runtime based on god knows what, and also not document whet they do there,  
The issue is that some combination of those things dosn't work properly with the new infra, and i caould not  determine where is the actual code for the test , forget trying to debug to fix it , like i've done for everyone else. I filed a bug and assigned it to Larry to fix. the fun thing is the code coverage is not affected byt those tests being live only.

File: sdk/core/azure-core-test/CMakeLists.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/core/azure-core-test/CMakeLists.txt:31]: done

File: sdk/core/azure-core-test/CMakeLists.txt [hunk 28]
```diff
@@ -46,4 +43,5 @@ target_include_directories (azure-core-test-fw
 # make sure that users can consume the project as a library.
 add_library (Azure::Core::Test ALIAS azure-core-test-fw)
 target_link_libraries(azure-core-test-fw PRIVATE azure-core Azure::azure-identity gtest)
-create_map_file(azure-core-test-fw azure-core-test-fw.map)
\ No newline at end of file
+create_map_file(azure-core-test-fw azure-core-test-fw.map)
+
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/core/azure-core-test/CMakeLists.txt:47]: done

File: sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp (hunk 17 above)
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp:17]: i think it's because the <> vs "" it puts "" then <> and sorts within the category

File: sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp [hunk 29]
```diff
@@ -7,21 +7,19 @@
...
 #include <azure/identity/client_secret_credential.hpp>
-
 #include <chrono>
 #include <memory>
 #include <regex>
 #include <thread>
-
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/core/azure-core-test/inc/azure/core/test/test_base.hpp:24]: done

File: sdk/core/azure-core-test/inc/azure/core/test/test_proxy_manager.hpp [hunk 30]
```diff
@@ -0,0 +1,179 @@
+#pragma once
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/core/azure-core-test/inc/azure/core/test/test_proxy_manager.hpp:1]: done

File: sdk/keyvault/assets.json [hunk 31]
```diff
@@ -0,0 +1,6 @@
+{
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/keyvault/assets.json:1]: the test-proxy documentation recommends the assets file to be next to ci.yml file

File: sdk/core/azure-core-test/inc/azure/core/test/test_proxy_manager.hpp [hunk 32]
```diff
@@ -0,0 +1,179 @@
...
+#if defined(BUILD_TRANSPORT_WINHTTP_ADAPTER)
+#include <azure/core/http/win_http_transport.hpp>
+#endif
+#include <azure/core/http/policies/policy.hpp>
+#include <azure/core/internal/http/pipeline.hpp>
+
+// Used by recordPolicy and playback transport adapter.
+#if !defined(RECORDING_BODY_STREAM_SENTINEL)
+#define RECORDING_BODY_STREAM_SENTINEL "__bodyStream__"
+#endif
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/core/azure-core-test/inc/azure/core/test/test_proxy_manager.hpp:24]: done, removed

File: sdk/core/azure-core-test/inc/azure/core/test/test_proxy_policy.hpp [hunk 33]
```diff
@@ -11,52 +11,40 @@
 #include <memory>
 #include <string>
 
+#include "azure/core/test/network_models.hpp"
+#include "azure/core/test/test_proxy_manager.hpp"
 #include <azure/core/context.hpp>
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/core/azure-core-test/inc/azure/core/test/test_proxy_policy.hpp:16]: done

File: sdk/core/azure-core-test/src/test_proxy_manager.cpp [hunk 34]
```diff
@@ -0,0 +1,309 @@
...
+    return Azure::Core::Test::TestMode::PLAYBACK;
+  }
+  else if (Azure::Core::_internal::StringExtensions::LocaleInvariantCaseInsensitiveEqual(
+               value, "LIVE"))
+  {
+    return Azure::Core::Test::TestMode::LIVE;
+  }
+
+  // unexpected variable value
+  throw std::runtime_error("Invalid environment variable: " + value);
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/core/azure-core-test/src/test_proxy_manager.cpp:45]: done

File: sdk/core/azure-core-test/inc/azure/core/test/test_proxy_policy.hpp (hunk 9 above)
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/core/azure-core-test/inc/azure/core/test/test_proxy_policy.hpp:74]: done

File: sdk/attestation/azure-security-attestation/test/ut/policygetset_test.cpp (hunk 14 above)
Comment (gearama) [reply sdk/attestation/azure-security-attestation/test/ut/policygetset_test.cpp:408]: yes , the tests in attesteation are overengneered to the point where i couldn't find the code that it executes durin tests in order to debug ans see why i had issues running the code. I had to move it to liveonly. i filed a bug to Larry to look into this  when he gets back.

File: sdk/core/azure-core-test/inc/azure/core/test/test_context_manager.hpp (hunk 16 above)
Comment (gearama) [reply sdk/core/azure-core-test/inc/azure/core/test/test_context_manager.hpp:27]: the test context uses the assets path in order to put toghether the start playback/record request. the idea is that you can override the GetAssetsPath to another location in the service testbase implementation. When the test context is initialized the assets path is set , then later on cosumed when each test is set to run, for each test you need to call startplayback , with the test name and the assets file

File: sdk/core/azure-core-test/inc/azure/core/test/test_proxy_manager.hpp [hunk 35]
```diff
@@ -0,0 +1,179 @@
...
+     * Sets proxy to start PLAYBACK test.
+     *
+     */
+    void SetStartPlaybackMode();
+
+    /**
+     * Sets proxy to stop PLAYBACK test.
+     *
+     */
+    void SetStopPlaybackMode();
```
[the remark this answers is not in the document]
Comment (gearama) [reply sdk/core/azure-core-test/inc/azure/core/test/test_proxy_manager.hpp:162]: done

**Review by benbp — APPROVED** [review]
EngSys configs look good.

File: sdk/core/azure-core-test/src/test_proxy_policy.cpp [hunk 36]
```diff
@@ -0,0 +1,83 @@
...
+  redirectRequest.SetHeader("x-recording-id", recordId);
+
+  if (m_testProxy->IsRecordMode())
+  {
+    // RECORDING mode
+    redirectRequest.SetHeader("x-recording-mode", "record");
+  }
+  else if (m_testProxy->IsPlaybackMode())
+  {
+    // RECORDING mode
```
Comment (RickWinter) [inline sdk/core/azure-core-test/src/test_proxy_policy.cpp:78]:
```suggestion
    // Playback mode
```

**Review by RickWinter — APPROVED** [review]

File: eng/scripts/Start-TestProxy.ps1 (hunk 25 above)
[the remark this answers is not in the document]
Comment (ahsonkhan) [reply eng/scripts/Start-TestProxy.ps1]: I recommend adding a link to the docs as a comment in source for future editors.
